IdentityCore/src/util/NSMutableDictionary+MSIDExtensions.h

```diff
@@ -26,5 +26,6 @@
 @interface NSMutableDictionary (MSIDExtensions)
 
 - (void)msidSetObjectIfNotNil:(id)dictObject forKey:(id<NSCopying>)dictKey;
+- (void)msidSetNonEmptyString:(NSString *)stringObj forKey:(id<NSCopying>)dictKey;
```

Comment:
Q: do we really need this to be different to the above? I think we can simply modify above method to achieve what we want.